### ISSUE TITLE
BUG:  Levure Tries to load a .git file as a library stack

### ISSUE THREAD

**Issue by macMikey** [trevordevore/levure#14]
I have created a repo for my libraries.  I then added them into this project, targeting the app/libraries folder.  Doing that added .git, .gitignore, and .gitmodules to the folder.  Levure then tries to load those files.  The same thing is going to happen, I would bet, for the readme.md and license.md files

I have a feeling that also means that in the helpers directory, files that are not scripts (I have a variety of secondary files that I want to include with my project) will suffer the same fate.

I understand that I could individually reference each of my stacks/SOS's in the yaml file, but I have a different idea that would be better:  START USING STACK should not reference files that do not have either the .livecodescript or .livecode suffix.

![screen shot 2017-03-19 at 20 05 08](https://cloud.githubusercontent.com/assets/10867016/24086113/856eca94-0cdf-11e7-8649-262ae3582adf.png)

**Comment by macMikey** [comment]
Follow-up on that:
Other invisible files, especially .DS_Store will be present in most folders, which will also cause a problem.

ALSO:
Is the traversal recursive?  If I have a folder inside of Libraries, will any stacks in there be loaded into the list?

**Comment by macMikey** [comment]
The line I would suggest changing is #1444, in addFolderToFilesArray
```
if isFilenameUniqueInFilesArray(tFile, xFilesA) then
```
I would change it to
```
if  (tFile ends with ".livecode OR tFile ends with ".livecodescript") AND (isFilenameUniqueInFilesArray(tFile, xFilesA)) then
```

**Comment by macMikey** [comment]
ALSO, you might want to have a try/catch before you start using one of these files.  I found one library that I had forgotten to add the "script" header to, and Levure brought up the SE with the error "Stack does not exist", which wasn't helpful until I did some digging.

**Comment by macMikey** [comment]
Pull request 15 filed.

**Comment by trevordevore** [comment]
This should be fixed in commit a3789f0be1b55674db9f70517a8d4fe0edf9a262.

This shouldn't be a problem with helpers as helpers don't allow you to point to a folder of libraries to load. You can only point to a specific files to load as libraries/behaviors/etc.
